woocommerce/woocommerce-ios
# [Mobile Payments] Upsell Card Reader Banner in App Settings

Closes: #7250 

⛔ Do not merge until [b62353706c0cb0c8f93254e7c402ebbecd983f30](https://github.com/woocommerce/woocommerce-ios/commit/599cd6bb74e6410506cbce5c0c98508e8fea2814) is reverted: This is just to make testing easier. With that test code we:
- Reset the banner policy show when starting the app, so it can be shown again even if the user dismissed it before.
- Set the remind me later to 20 seconds instead of 14 days, so it can be tested faster.



### Description

With this PR we want to add the upsell card readers feature banner in settings as well, so the users can be reminded of this new functionality when opening the settings. 
This banner is basically the same we implemented and show in the Payment Methods screen, with the only difference that we do not show the call to action button here because we already have the In-Person Payments settings row right below. Please notice that the banner show policy is the same as the one shown in the Payment Methods screen, so if it is dismissed (forever or just to be reminded later) there, it won't be shown in the settings. We also have the same flow, when we the user presses on [X] we show an alert to choose between dismissing  it forever or just being reminded later on.

### Testing instructions

Using a US or Canada store:

1. Navigate to Settings (Menu tab, settings icon)
2. Observe that the Upsell Card Reader banner is shown, with a x button to dismiss.
3. Tap the x button and then the `Remind me later` option – observe that the card is dismissed
4. Go to settings again within the remind time, and observe that the banner doesn't show there.
5. Go back from settings and open it again after the remind time elapses, observe that the card is shown.

Repeat the above test using `Don't show again`, and observe that the card is not shown even after the remind time elapses.

To reset and re-test, either delete and reinstall the app, or restart the app using the test hack implemented in 599cd6bb74e6410506cbce5c0c98508e8fea2814 to clear the stored dismissal. This is not part of the feature, and will be removed before it is merged.

Please make sure that tracking for show and dismiss events is properly triggered, with the new source value "settings".

### Screenshots


https://user-images.githubusercontent.com/1864060/179937915-64ad836e-cacb-4e9e-bdcb-8fb30ad39013.mp4




---
